File: flightpub-api/src/main/java/seng3150/team4/flightpub/domain/models/HolidayPackageBooking.java
package seng3150.team4.flightpub.domain.models;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;

@Table(name = "HolidayPackageBooking")
@Entity
@NoArgsConstructor
@Getter
@Setter
public class HolidayPackageBooking implements IEntity {
  @Column(name = "Id")
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @ManyToOne
  @JoinColumn(name = "UserId")
  private User user;

  @ManyToOne
  @JoinColumn(name = "HolidayPackageId", referencedColumnName = "Id")
  private HolidayPackage holidayPackage;

  @Column(name = "DateBooked")
  private LocalDateTime dateBooked;

  @ManyToOne
  @JoinColumn(name = "PaymentId", referencedColumnName = "Id")
  private Payment payment;

  @ManyToOne
  @JoinColumn(name = "BookingId", referencedColumnName = "Id")
  private Booking booking;
}

File: flightpub-ui/src/models/HolidayPackageBooking.ts
import { HolidayPackage } from './HolidayCardProps';
import { User } from './User';
import { Booking } from './Booking';
import { PaymentType } from './SavedPaymentTypes';

export interface HolidayPackageBooking {
  id: number;
  user: User;
  dateBooked: Date;
  payment: PaymentType;
  booking: Booking;
  holidayPackage: HolidayPackage;
}
